src/parsimmon.js

```diff
@@ -396,14 +396,16 @@ _.then = function(next) {
 
 _.many = function() {
   var self = this;
-
   return Parsimmon(function(input, i) {
     var accum = [];
     var result = undefined;
-
+    
     for (;;) {
       result = mergeReplies(self._(input, i), result);
       if (result.status) {
+        if (i === result.index) {
+          return makeFailure(i, 'Infinite loop has been detected in many() parser');
+        }
```

Comment:
hmm, should this use makeFailure, or `throw new Error(...)`. it isn't a failed parse? it's a broken parser? i don't know.

Reply:
@minamorl Do you still have time to change to `throw new Error(...)`?